seqr/views/apis/anvil_workspace_api.py

```diff
@@ -0,0 +1,118 @@
+        project = Project.objects.filter(name=name)
+        if project:
+            return redirect('/project/{}'.format(project.guid))
+        else:
+            return redirect('/create_project_from_workspace/{}/{}'.format(namespace, name))
+    return {"Error: Workspace {}/{} doesn't exist or misses required permissions (e.g. WRITER, OWNER, or PROJECT_OWNER)"
+                .format(namespace, name)}
+
+
+@login_required(login_url=API_LOGIN_REQUIRED_URL)
+def create_project_from_workspace(request, namespace, name):
+    """
+    Create a project when a cooperator requests to load data from an AnVIL workspace.
+
+    :param request: Django request object
+    :param namespace: The namespace (or the billing account) of the workspace
+    :param name: The name of the workspace. It also be used as the project name
+    :return the projectsByGuid with the new project json
+    """
+    # Validate that the current user has logged in through google and has one of the valid can_edit levels of
+    #  access on the specified workspace
+    project = Project.objects.filter(name = name)
+    if project:
+        error = 'Project {} exists.'.format(name)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    if not _workspace_can_edit(request.user, namespace, name):
+        return create_json_response({
+            'projectsByGuid': {},
+        })
+
+    # Validate all the user inputs from the post body
+    request_json = json.loads(request.body)
+
+    missing_fields = [field for field in ['genomeVersion', 'uploadedFileId'] if not request_json.get(field)]
+    if missing_fields:
+        error = 'Field(s) "{}" are required'.format(', '.join(missing_fields))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    if not request_json.get('agreeSeqrAccess'):
+        error = 'Must agree to grant seqr access to the data in the associated workspace.'
+        return create_json_response({'error': error}, status = 400, reason = error)
+
+    # Add the seqr service account to the corresponding AnVIL workspace
+    if add_service_account(request.user, namespace, name):
+        # Create a new Project in seqr
+        project_args = {
+            'name': name,
+            'genome_version': request_json['genomeVersion'],
+            'description': request_json.get('description', ''),
+            'workspace_namespace': namespace,
+            'workspace_name': name,
+        }
+
+        project = create_model_from_json(Project, project_args, user = request.user)
+
+        # Add families/individuals based on the uploaded pedigree file
+        try:
+            json_records = load_uploaded_file(request_json['uploadedFileId'])
+        except Exception as ee:
+            error = "Uploaded pedigree file is missing or other exception: {}".format(str(ee))
+            return create_json_response({'error': error}, status = 400, reason = error)
```

Comment:
you should actually move the file parsing above the code that creates the project model. Otherwise, what happens is as follows:

- User hits submit with an invalid file attached
- seqr creates the project associated with the namespace/name
- validation of the file fails. No families/ individuals are added to the new project. No notification is sent to us that someone requested data loading. An error is shown to the user
- user uploads a valid file and hits submit again
- An error is shown to the user saying that a project with this name already exists. There is no way for the user to successfully submit the request data loading form anymore